bk-api/bk_api.py

```diff
@@ -1316,102 +1253,95 @@ def get(self):
         Returns:
             dict: end-point id, private key, public key and certificate
         """
-        node_id = request.args.get("id", type=str)
+        logger.info("registration request: %r", request.args)
+
+        node_id = request.args.get("node_id", type=str)
+
         if not node_id:
-            logger.debug("Registration failed: id missing")
-            return f"Error: id missing\n", 500
+            return "error: node id must be provided.\n", HTTPStatus.BAD_REQUEST
 
-        logger.debug("Register user [{}]".format(node_id))
+        if not valid_node_id(node_id):
+            return "error: invalid node_id. (node ids must be between 6 and 32 [a-z0-9] characters.)\n", HTTPStatus.BAD_REQUEST
 
+        beehive_id = request.args.get("beehive_id", DEFAULT_BEEHIVE, type=str)
 
-        if DEFAULT_BEEHIVE:
+        if beehive_id:
             # first check the default beehive exists
```

Comment:
this is not checking the default beehive now...

Reply:
Good catch - that comment is wrong now. I'll remove it.